src/fu-device.c

```diff
@@ -468,7 +471,10 @@ fu_device_add_child_by_type_guid (FuDevice *self, GType type, const gchar *guid)
 	fu_device_add_guid (child, guid);
 	if (fu_device_get_platform_id (self) != NULL)
 		fu_device_set_platform_id (child, fu_device_get_platform_id (self));
+	if (!fu_device_probe (child, error))
+		return FALSE;
 	fu_device_add_child (self, child);
```

Comment:
Would you consider to probe after the child is added instead?  I'd like to be able to reference the parent when probe is run.

Reply:
Should the child be added if the probe fails?

Reply:
Probably not.  How about instead running probe within `fu_device_add_child`?  It wouldn't be run twice because of `done_probe`.

Reply:
`fu_device_add_child()` doesn't take a GError, and so we'd have no way to know if the child actually got added or not. Back a bit: why does the probe need to know about the parent?

Reply:
So that it can gather something from the parent that's used in `open`.  I guess the alternative is to move that into open if it's not set to grab it at that time.

Reply:
Well open runs before setup, so open would fail.  I'll move it to open then and this PR is fine.